features/nanostack/coap-service/source/coap_security_handler.c

```diff
@@ -388,17 +420,35 @@ int coap_security_handler_connect_non_blocking(coap_security_t *sec, bool is_ser
         mbedtls_ssl_conf_handshake_timeout(&sec->_conf, timeout_min, timeout_max);
     }
 
-    mbedtls_ssl_conf_rng(&sec->_conf, mbedtls_ctr_drbg_random, &sec->_ctr_drbg);
+#if !defined(MBEDTLS_SSL_CONF_RNG)
+#if defined(MBEDTLS_CTR_DRBG_C)
+    mbedtls_ssl_conf_rng(&sec->_conf, mbedtls_ctr_drbg_random, &sec->_drbg);
+#elif defined(MBEDTLS_HMAC_DRBG_C)
+    mbedtls_ssl_conf_rng(&sec->_conf, mbedtls_hmac_drbg_random, &sec->_drbg);
+#endif
```

Comment:
Similarly here #else ?

Reply:
No, we don't want to pollute code with dead code

Reply:
So only one of those 2 conditions is always guaranteed? If that is the case then the #elif should be a #else ? Otherwise you have unexpected code execution....

Reply:
Same point as [above](https://github.com/ARMmbed/mbed-os/pull/12479#discussion_r400106285).

Reply:
Fixed